lib/screens/editor/versions.dart

```diff
@@ -0,0 +1,398 @@
+  }
+
+  Table _buildTable(Version version) {
+    return Table(
+      border: TableBorder.all(
+        width: 1,
+        color: Colors.grey,
+        borderRadius: BorderRadius.all(
+          Radius.circular(3),
+        ),
+      ),
+      children: [
+        for (final tag in version.tagChanges.entries)
+          TableRow(
+            // TODO: these BoxDecorations slightly clip outside of the table BorderRadius
+            decoration: BoxDecoration(
+              color: tag.value.getColor(),
+            ),
+            children: [
+              Padding(
+                padding: const EdgeInsets.all(6),
+                child: Text(
+                  tag.key,
+                  style: TextStyle(
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+              ),
+              Padding(
+                padding: const EdgeInsets.all(6),
+                child:
+                    Text(!tag.value.noChange ? tag.value.oldValue ?? '' : ''),
+              ),
+              Padding(
+                padding: const EdgeInsets.all(6),
+                child: Text(tag.value.newValue ?? ''),
+              )
+            ],
+          ),
+      ],
+    );
+  }
+
+  Card _buildCard(Version version, AppLocalizations loc) {
+    return Card(
+      margin: EdgeInsets.only(top: 12, right: 12, left: 12),
+      child: Padding(
+        padding: const EdgeInsets.all(10),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Padding(
+              padding: const EdgeInsets.only(bottom: 8),
+              child: Text(
+                version.isLocal
+                    ? loc.versionsLocalChanges
+                    : loc.versionsVersionNumber(version.number),
+                style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16),
+              ),
+            ),
+            if (!version.isLocal)
+              Padding(
+                padding: const EdgeInsets.only(bottom: 8.0),
+                child: Text(
+                  loc.versionsVersionMeta(
+                    version.user,
+                    DateFormat.yMMMMd().add_Hm().format(version.timestamp),
+                  ),
+                ),
+              ),
+            if (version.comment != null)
+              Padding(
+                padding: const EdgeInsets.only(bottom: 8.0),
+                child: Text(
+                  version.comment!,
+                  style: TextStyle(fontStyle: FontStyle.italic),
+                ),
+              ),
+            version.noTagChange
+                ? Text(loc.versionsNoTagChanges)
+                : _buildTable(version),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final loc = AppLocalizations.of(context)!;
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(loc.versionsTitle(widget.fullRef)),
+        actions: [
+          IconButton(
```

Comment:
Is this button needed now? I thing this page includes all the info from osm.org, maybe without geometry, but at this point the user knows where the object is.

Reply:
I am \_weakly\_ in favour of keeping it. It does make it a bit easier to send someone a link to the history page for an element, and it doesn't add too much visual clutter.

If you think it is better without then I'm not opposed :)

Reply:
Well, there's still a "share" button on the tags panel, so again this one is redundant. I'd vote for removing it.

Reply:
Makes sense, removed in 1446e00a